From 5d562e9fee09d6682dd7494a0e7252dae2b874ac Mon Sep 17 00:00:00 2001
From: Laur59 <148864407+Laur59@users.noreply.github.com>
Date: Tue, 20 Jan 2026 13:46:00 +0200
Subject: [PATCH 2/3] build: Add extra LLVM tools to build list

---
 arm-software/embedded/CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arm-software/embedded/CMakeLists.txt b/arm-software/embedded/CMakeLists.txt
index 9db60e1d0181..ec588216cb76 100644
--- a/arm-software/embedded/CMakeLists.txt
+++ b/arm-software/embedded/CMakeLists.txt
@@ -151,7 +151,7 @@ set(PARALLEL_LIB_BUILD_LEVELS
 option(
     ENABLE_QEMU_TESTING
     "Enable tests that use QEMU. This option is ON by default."
-    ON
+    OFF
 )
 option(
     ENABLE_FVP_TESTING
@@ -231,6 +231,8 @@ set(LLVM_DISTRIBUTION_COMPONENTS
     llvm-strings
     llvm-strip
     llvm-symbolizer
+    scan-build
+    clang-tidy
     LTO
     CACHE STRING ""
 )
@@ -243,7 +245,7 @@ set(LLVM_TOOLCHAIN_DISTRIBUTION_COMPONENTS
     CACHE STRING "Components defined by this CMakeLists that should be
 installed by the install-llvm-toolchain target"
 )
-set(LLVM_ENABLE_PROJECTS clang;lld CACHE STRING "")
+set(LLVM_ENABLE_PROJECTS clang;lld;clang-tools-extra CACHE STRING "")
 set(LLVM_TARGETS_TO_BUILD AArch64;ARM CACHE STRING "")
 set(LLVM_DEFAULT_TARGET_TRIPLE aarch64-unknown-linux-gnu CACHE STRING "")
 set(LLVM_APPEND_VC_REV OFF CACHE BOOL "")
-- 
2.50.1 (Apple Git-155)

